cmd/build/v2/services.go

```diff
@@ -118,3 +116,37 @@ func getToBuildTag(manifestName, svcName string, b *model.BuildInfo) string {
 	}
 	return ""
 }
+
+func channelToSet[T comparable](ch chan T) map[T]bool {
+	set := make(map[T]bool)
+	for value := range ch {
+		set[value] = true
+	}
+	return set
+}
+
+func sliceToSet[T comparable](slice []T) map[T]bool {
```

Comment:
I wouldn't include a generic function if it is not needed. We are jut using this function to convert a single type ([]string). I would try to keep things simple as much as we can.

If you want to go forward with the generics, I think all these functions should be moved to its own package instead of making them private for the `v2` package

Reply:
I wasn't meaning a new library, I was meaning a package in this project